src/engine/BMGameAction.php

```diff
@@ -660,17 +668,52 @@ protected function message_size_change($preInfo, $postInfo) {
      */
     protected function message_recipe_change($preInfo, $postInfo, $showFrom = TRUE) {
         $message = '';
+        $fromMessage = '';
 
-        if ($preInfo['recipe'] != $postInfo['recipe']) {
-            $message = 'recipe changed';
+        if ($showFrom) {
+            $fromMessage = ' from ' . $preInfo['recipe'];
+        }
 
-            if ($showFrom) {
-                $message .= ' from ' . $preInfo['recipe'];
-            }
+        $hasJustBeenMoody = array_key_exists('hasJustBeenMoody', $postInfo);
 
-            $message .= ' to ' . $postInfo['recipe'];
-        }
 
+        if (!$hasJustBeenMoody) {
+            if ($preInfo['recipe'] == $postInfo['recipe']) {
+                // no change in recipe, do nothing
+            } else {
+                // non-mood change in recipe
+                $message .= 'recipe changed' .
+                            $fromMessage .
+                            ' to ' . $postInfo['recipe'];
+            }
+        } else {
+            if ($preInfo['recipe'] == $postInfo['hasJustBeenMoody']) {
+                if ($postInfo['hasJustBeenMoody'] == $postInfo['recipe']) {
+                    // no non-mood change in recipe, followed by a mood change to the same recipe
+                    $message .= 'recipe remained as ' .
```

Comment:
FWIW, "recipe remained as fo(O)" sounds unnatural to me.  I think "recipe remained fo(O)" is more natural.